src/mlio-py/mlio/contrib/insights/column_statistics.h

```diff
@@ -45,13 +45,26 @@ class Column_analysis {
         return static_cast<std::size_t>(std::round(str_cardinality_estimator_.estimate()));
     }
 
+public:
+    double estimate_median_approx() const
+    {
+        if (numeric_column_sample.empty()) {
+            return std::nan("");
+        }
+        std::vector<double> v = numeric_column_sample;
+        size_t n = v.size() / 2;
+        std::nth_element(v.begin(), v.begin() + n, v.end());
+        return v[n];
+    }
+
 public:
     std::string column_name;
 
     std::size_t rows_seen{};
     std::size_t numeric_count{};
     std::size_t numeric_nan_count{};
     std::size_t numeric_finite_count{};
+    std::vector<double> numeric_column_sample{};
```

Comment:
Does this need to be a public member? Does it carry any meaning by itself outside of the `estimate_median_approx()` function?

Reply:
Will move to private